packages/repository/src/mixins/repository.mixin.ts

```diff
@@ -97,9 +97,11 @@ export function RepositoryMixin<T extends Class<any>>(superClass: T) {
      * }
      * ```
      */
-    dataSource(dataSource: juggler.DataSource, name?: string) {
+    dataSource(dataSource: Class<juggler.DataSource>, name?: string) {
```

Comment:
I feel LB4 apps should not be required to always define a new DataSource class. Can we support dual usage allowing both a DataSource instance and a DataSource class?

```ts
dataSource(dataSource: juggler.DataSource | Class<juggler.DataSource>, name? string): void {
  if (typeof juggler.DataSource === 'function') {
    // handle the Class case
  } else {
    // handle the instance case
  }  
}
```

Thoughts?

Reply:
Is there a use case you have in mind where someone will prefer to create an instance instead of a DataSource class?

Reply:
```ts
import {Application} from '@loopback/core';
import {juggler} from '@loopback/repository';

const app = new Application();
app.dataSource(new juggler.DataSource('db', {connector: 'memory'}));
// etc.
```

While this is not a typical usage in LB4 apps scaffolded by our CLI tooling, it makes it much much simpler to write short scripts reproducing certain behavior (e.g. a bug).